types/bl/index.d.ts

```diff
@@ -1,25 +1,29 @@
-// Type definitions for BufferList v4.1
+// Type definitions for BufferList v5.0
```

Comment:
From the readme: 


> If you intend to continue updating the older version of a library's type declarations, you may create a new subfolder (e.g. /v2/) named for the current (soon to be "old") version, and copy existing files from the current version to it.

Reply:
@amcasey I'm one of the maintainers of this package; not sure if your suggestion is worthwhile only because @rvagg has expressed interest in moving this into `bl` as a first-class citizen and existing users can pin the @types/bl package version they need in the mean time.

Reply:
@benallfree Creating an old-version directory is very optional and I was simply asking whether it would be beneficial for this package.  Having said that, I'm not sure I understand how the comment you linked answers the question - first class types would be great but users of v4 would still consume `@types/bl` and might still want bug fixes for those types.  If usage of v4 is limited or if the types are simple enough to be "obviously correct", then it's probably not worthwhile.

Reply:
@amcasey Sorry yes I meant to say that I see your point and it could be valuable to maintain older versions. There are already known issues/shortcomings with the v4 typings, for example.

I was attempting to balance that against how much longer this package would need to be maintained, but now that I think about it more, I realize that this @types/bl package can really never be deprecated because older versions of bl still need it.

So I agree with you now :) We should folder-ize this package.

Reply:
I don't think this is necessary, at least not yet - the bump from v4 to v5 happened because the underlying [buffer](https://www.npmjs.com/package/buffer) browser polyfill jumped from v5 to v6, removing ie11 and Safari < 10 support, not because there was any API breakage.

Would it be better to hold off creating folders until there's a release that breaks the API, give that if we create folders now the contents of v4 and v5 would be identical?

Reply:
@achingbrain Hmm I agree with your point too! I'm good either way 👍

Reply:
@achingbrain I'm not sure I understand your suggestion.  I think you're saying that users currently consuming bl@4 should start consuming @types/bl@5 because the APIs are compatible?

Reply:
To reiterate though, I'm fine with not creating a v4 directory.

Reply:
@amcasey yes, that’s what I’m saying - the APIs are compatible.